Astronomy/View Controllers/PhotosCollectionViewController.swift

```diff
@@ -64,13 +64,81 @@ class PhotosCollectionViewController: UIViewController, UICollectionViewDataSour
     
     private func loadImage(forCell cell: ImageCollectionViewCell, forItemAt indexPath: IndexPath) {
         
-        // let photoReference = photoReferences[indexPath.item]
-        
         // TODO: Implement image loading here
+        
+        let photoReference = photoReferences[indexPath.item]
+        let photoURL = photoReference.imageURL.usingHTTPS
+        
+        if let cacheImage = cache.value(for: photoReference.id){
+            
+            cell.imageView = UIImageView(image: UIImage(data: cacheImage))
```

Comment:
You should set `imageView`s `image` not itself 
```suggestion
            cell.imageView.image = UIImage(data: cacheImage)
```